lib/validation/schema.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Validation.Schema do
+  @moduledoc """
+  A Schema is a collection of rules that each are applied to the given params.
+  The result of evaluating a params map against a schema is a %Result{} struct.
+  """
+
+  alias Validation.Result
+
+  defstruct [
+    val: nil,
+    meta: []
+  ]
+
+  def build(rules) do
+    val = fn params ->
+      result = %Result{data: params}
+      rules
+      |> Enum.reduce(result, fn rule, result ->
```

Comment:
I think you are absolutely right about this.
Making Rule the module that knows how to apply a rule to a result seems like the only reasonable thing to do :)
Adding an `apply` to Schema is good for consistency.

Will fix,